lib/puppet/ssl/inventory.rb

```diff
@@ -40,7 +44,10 @@ def rebuild
   def serials(name)
     return [] unless Puppet::FileSystem.exist?(@path)
 
-    File.readlines(@path).collect do |line|
+    # RFC 5280 says the cert subject may contain UTF8 - https://www.ietf.org/rfc/rfc5280.txt
+    # Note however that Puppet generated SSL files must only contain ASCII characters
+    # based on the validate_certname method of Puppet::SSL::Base
+    File.readlines(@path, :encoding => Encoding::UTF_8).collect do |line|
```

Comment:
@adrienthebo has noted that this is using `File.readlines` instead of a `Puppet::FileSystem` variation.

This is because `Puppet::FileSystem` does not have a `readlines` defined in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_system.rb - its on the TODO list eventually